src/collections/src/NICollectionViewModel.m

```diff
@@ -176,6 +176,46 @@ - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
   return nil;
 }
 
+#pragma mark - NSFastEnumeration
+
+- (NSUInteger)countByEnumeratingWithState:(NSFastEnumerationState *)state objects:(__unsafe_unretained id  _Nonnull *)buffer count:(NSUInteger)len {
+  // Named indexes into state->extra.
+  static const unsigned int kExtraSectionIndex = 0;
+  static const unsigned int kExtraItemIndex = 1;
+
+  // Initial setup is needed when state is zero.
+  if (state->state == 0) {
+    state->state = 1;
+    state->itemsPtr = buffer;
+    state->mutationsPtr = &_mutationCount;
+  }
+
+  NSArray<NICollectionViewModelSection *> *sections = self.sections;
+
+  NSUInteger dstIdx = 0;
```

Comment:
This is common shorthand for the "destination index" we're writing to in the output buffer.